frappe/integrations/doctype/s3_backup_settings/s3_backup_settings.py

```diff
@@ -34,11 +34,27 @@ def validate(self):
 		except ClientError:
 			frappe.throw(_("Invalid Access Key ID or Secret Access Key."))
 
+		bucket_name_exist = False
 		try:
-			conn.create_bucket(Bucket=bucket_lower, CreateBucketConfiguration={
-				'LocationConstraint': self.region})
-		except ClientError:
-			frappe.throw(_("Unable to create bucket: {0}. Change it to a more unique name.").format(bucket_lower))
+			response = conn.head_bucket(Bucket=bucket_lower)
+			# The operation returns a 200 OK if the bucket exists and you have permission to access it.
+			bucket_name_exist = True
+		except ClientError as e:
+			# If a client error is thrown, then check that it was a 403 error or 404 error.
+			error_code = e.response['Error']['Code']
+			if error_code == '403':
+				frappe.throw(_(f"403 Forbidden. Do not have permission to access this bucket: {bucket_lower}"))
+			elif error_code == '404':
+				# Bucket not found, set bucket_name_exist flag to False to try create bucket
+				bucket_name_exist = False
```

Comment:
instead of having a flag for checking if bucket name exists, we can actually just create the bucket here and save some lines of code.
```suggestion
				conn.create_bucket(Bucket=bucket_lower, CreateBucketConfiguration={
					'LocationConstraint': self.region})
```

there's also no need to throw:

```python
except ClientError:
	frappe.throw(_("Unable to create bucket: {0}. Change it to a more unique name.").format(bucket_lower))
```

since `head_bucket()` should already tell you if you have access to the bucket if it exists. so the exception is pretty redundant now.